src/sentry/snuba/metrics_layer/query.py

```diff
@@ -0,0 +1,91 @@
+from typing import Sequence
+
+from snuba_sdk import Request
+from snuba_sdk.metrics_query import MetricsQuery
+
+from sentry.models import Project
+from sentry.sentry_metrics.utils import resolve_weak, string_to_use_case_id
+from sentry.snuba.metrics.fields.base import _get_entity_of_metric_mri, org_id_from_projects
+from sentry.snuba.metrics.naming_layer.mapping import get_mri, get_public_name_from_mri
+from sentry.snuba.metrics.utils import to_intervals
+
+
+def run_query(request: Request) -> None:
```

Comment:
I am very happy to see some efforts to improve `get_series` I would just be curious to know if they are part of the effort that will be driven by SNS to evolve the metrics layer and eventually replace it with the new experimental implementation.

Reply:
Yes this is the new implementation. Or at least the first version of it.